AirCasting/Utils/AveragingService.swift

```diff
@@ -122,35 +125,60 @@ final class AveragingService: NSObject, ObservableObject {
        return values.map({ $0.value }).reduce(0.0, +) / Double(values.count)
     }
     
+    private func averagedMeasurementFrom(chunk: ArraySlice<MeasurementEntity>, window: AveragingWindow, measurementCount: Int) -> [MeasurementEntity] {
+        if chunk.count == measurementCount {
+            /// https://github.com/apple/swift-algorithms/blob/main/Guides/Chunked.md
+            /// For integer types, any remainder of the division is discarded so we need to add 1 to elementsInChunk/2 to get the middle value.
+            let middleIndex = chunk.index(chunk.startIndex, offsetBy: (measurementCount/2 + 1))
```

Comment:
I think this should be `measurementCount/2 - 1` if we want a middle measurement cause 12/2 + 1 = 7 and the middle is 5

Reply:
Actually, it should be just `measurementCount/2` for odd numbers like `firstThresholdWindow` and what you said for even numbers like in the case of `secondThresholdWindow`. Thanks!

Reply:
@mturnau when you have an even number of measurements in the chunk which one do you consider the middle measurement?

Reply:
btw. note that this parameter is the offset, not an index. So it actually "starts" from 1. We can also create a simple extension for this so it can be reused and/or covered byt unit tests:

```Swift
extension RandomAccessCollection {
    var middleItems: (first: Element, second: Element?) {
        let firstIdx = index(startIndex, offsetBy: count / 2)
        guard count.isEven else {
            return (first: self[firstIdx], second: nil)
        }
        return (first: self[index(firstIdx, offsetBy: -1)], second: self[firstIdx])
    }
}

extension FixedWidthInteger {
    var isEven: Bool {
        self % 2 == 0
    }
}
```

output examples:
```Swift
// Odd, we have a single middle item
let odd = [0,1,2,3,4,5,6,7,8]
print(odd.middleItems) // (first: 4, second: nil)
// Even - we have 2 middle items
let even = [0,1,2,3,4,5]
print(even.middleItems) // (first: 2, second: Optional(3))
// Empty
let empty: [Int] = []
print(empty.middleItems) // nil
// Signgle item
let single = [42]
print(single.middleItems) // (first: 42, second: nil)
// Two items
let double = [42, 24]
print(double.middleItems) // (first: 42, second: Optional(24))
```

Reply:
We don't want to have two middle items, we need just one to get all the measurement data apart from value. 
But I can make an extension for that anyways.

Reply:
We don't want to have two middle elements, just one. Added extensions.